### PR TITLE
refactor: fix clippy issues for near-o11y

### PR DESCRIPTION
**Pull request by pugachAG**
Fix the issues reported by `cargo clippy -p near-o11y` as part of #8145

### CODE REVIEW

**Comment by nagisa** [comment]
IMO we should decide which clippy lints we're enabling _first_, set them to `warn` and only then start fixing the issues, possibly removing the lints as we go.

In particular I would probably advocate for disabling the lint that necessitated the `n /= 10` change. While neat that it is possible to write this in a terser way, I think the original was tad a bit more readable there.

Another approach that I have been using in the past is set up clippy without lints at all, and then enable them one-by-one, fixing the specific lints for the entire codebase at once. It then makes discussions about whether the lint is valuable for us or not much more… substantiated.

**Comment by nagisa** [comment]
I would advocate allowing all style category lints for the initial pass, actually. No reason to hinder the adoption with what are somewhat opinionated calls. Lets give time for people to get used to having clippy tell them about important stuff first. I think they’ll be much more open to majority of the style & pedantic lints once they know clippy is a highly valuable tool :)

**Comment by jakmeier** [comment]
Very good points, I like the idea of @nagisa to allow all style category lints to start with. Adding them one-by-one and discussing whether a particular lint is useful seems like a good approach. How about we allow all and warn/deny on `clippy::correctness` + `clippy::suspicious` at first?

File: core/o11y/benches/metrics.rs [hunk 1]
```diff
@@ -63,7 +63,7 @@ fn inc_counter_vec_with_label_values_stack_no_format(bench: &mut Bencher) {
             loop {
                 idx -= 1;
                 buf[idx] = b'0' + (n % 10) as u8;
-                n = n / 10;
+                n /= 10;
```
Comment (jakmeier) [inline core/o11y/benches/metrics.rs:66]: this also feels a bit odd to me, eitherway it's fine but personally I don't think `n /= 10`  is better than `n = n / 10`

**Review by jakmeier — APPROVED** [review]
I think I like all changes except for the `/=`, and even that doesn't personally bother me if we change it. :smile:  So I'll approve the PR, independent of how how we move forward with enabling clippy

**Comment by pugachAG** [comment]
@nagisa @jakmeier thanks for sharing your feedback on the overall approach. I agree that it makes sense to first enable clippy with the most useful lints (`clippy::correctness` + `clippy::suspicious` as was suggested by Jakob). Further I will work on gradually enabling parts of clippy rather than fixing all issues for every package.

I will merge with this PR while reverting the `n /= 10` change.

**Pull request merged**